crates/relayer/src/config.rs

```diff
@@ -50,6 +53,54 @@ impl Display for GasPrice {
     }
 }
 
+impl TryFrom<String> for GasPrice {
+    type Error = ConfigError;
+
+    fn try_from(price_in: String) -> Result<Self, Self::Error> {
+        // TODO: We split by `char::is_alphabetic` delimiter.
+        //      More robust parsing methods might be needed.
+        let spos = price_in.find(char::is_alphabetic);
+
+        match spos {
+            Some(position) => {
+                let (price_str, denom) = price_in.split_at(position);
+
+                let price = price_str
+                    .parse::<f64>()
+                    .map_err(|_| Error::invalid_gas_price(price_in.to_string()))?;
+
+                Ok(GasPrice {
+                    price,
+                    denom: denom.to_owned(),
+                })
+            }
+
+            None => Err(Error::invalid_gas_price(price_in)),
+        }
+    }
+}
+
+impl PartialOrd for GasPrice {
```

Comment:
Let's perhaps add a comment explaining why we don't implement `Ord` here, ie. because only gas prices for the same denom can be ordered relative to one another.